From e9786b91cd974fb4f68bfe14e76c9eeced9c0ac0 Mon Sep 17 00:00:00 2001
From: Tim Carney <tim.carney@cargt.com>
Date: Tue, 26 May 2026 12:11:19 -0500
Subject: [PATCH] board/cargt(imx93_00363): add 512MB@3733 timing option

---
 board/cargt/imx93_00363/spl-cargt.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/board/cargt/imx93_00363/spl-cargt.c b/board/cargt/imx93_00363/spl-cargt.c
index a99b037c97d..6ffd0cc7f3f 100644
--- a/board/cargt/imx93_00363/spl-cargt.c
+++ b/board/cargt/imx93_00363/spl-cargt.c
@@ -79,6 +79,7 @@ extern struct dram_timing_info cargt_lpddr4x_2gbyte_3733mts_timing;
 extern struct dram_timing_info cargt_lpddr4x_1gbyte_3200mts_timing;
 extern struct dram_timing_info cargt_lpddr4x_2gbyte_3200mts_timing;
 extern struct dram_timing_info cargt_lpddr4x_512mb_3200mts_timing;
+extern struct dram_timing_info cargt_lpddr4x_512mb_3733mts_timing;
 
 /*
  * Select a board-provided `dram_timing_info` based on requested
@@ -102,7 +103,9 @@ static struct dram_timing_info *select_dram_timing(uint16_t speed_mts,
 	if (speed_mts == 3733) {
 		if (size_mbytes == 2048)
 			return &cargt_lpddr4x_2gbyte_3733mts_timing;
-		/* no 512@3733 reference yet: fall back to 3200 512MB timing */
+		if (size_mbytes == 512)
+			return &cargt_lpddr4x_512mb_3733mts_timing;
+		/* otherwise fall back to 1GB@3733 */
 		return &cargt_lpddr4x_1gbyte_3733mts_timing;
 	}
 
